fawltydeps/packages.py

```diff
@@ -319,7 +354,8 @@ def lookup_packages(self, package_names: Set[str]) -> Dict[str, Package]:
 
 def resolve_dependencies(
     dep_names: Iterable[str],
-    custom_mapping_path: Optional[Path] = None,
+    custom_mapping_file: Optional[Path] = None,
+    custom_mapping: Optional[Dict[str, List[str]]] = None,
     pyenv_path: Optional[Path] = None,
     install_deps: bool = False,
 ) -> Dict[str, Package]:
```

Comment:
I would do it in a later PR. It may even require grouping of `Setting` object attributes. We have a lot of settings that are unrelated to resolving dependencies, such as verbosity, actions, etc.